x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/FieldAttribute.java

```diff
@@ -142,7 +142,9 @@ public int hashCode() {
 
     @Override
     public boolean equals(Object obj) {
-        return super.equals(obj) && Objects.equals(path, ((FieldAttribute) obj).path);
+        return super.equals(obj)
+            && Objects.equals(path, ((FieldAttribute) obj).path)
+            && Objects.equals(field, ((FieldAttribute) obj).field);
```

Comment:
Modifying this one scares me a bit.

Reply:
Instead of modifying this, we should create a copy in the esql project.

This may involve some yak shaving, but I'm happy to help with that!